utils/stm.stanza

```diff
@@ -221,26 +306,26 @@ defn create-pin-properties (json1:JSON) -> STMPinProperties :
  
   ;to-tuple(pin-properties)
   val stm-properties = STMPinProperties(stm-gen-pin, stm-pow-pin, to-tuple(pin-properties))
-  println("%_, %_, %_" % [ pin-name(generic-pin(stm-properties)), pin-name(power-pin(stm-properties)), pin-name(rows(stm-properties)[0])])
+  println("%_, %_, %_" % [ pin-name(generic-pin(stm-properties)), pin-name(power-pin(stm-properties)), pin(rows(stm-properties)[0])])
   stm-properties
 
+;=================================================
+;==================== Testing ====================
+;=================================================
 
 val JSON-TEST-FILE-LIST = to-hashset<String> $ [
-  "~/dev/jitx/mono_repo/jitx-client/generate-stm/stm-test.json"
+  "stm-test.json"
 ]
 
-defn main ():
-  for file in JSON-TEST-FILE-LIST do:
-    val input-json:JSON = parse $ slurp(file)
-    val new-bundles = create-bundles(input-json)
-    for bundle in new-bundles do:
-      println(name(bundle))
-
-    val pin-properties = create-pin-properties(input-json)
-    for pin-item in rows(pin-properties) do:
-      println("%_, %_, %_, %_, %_" % [pin-name(pin-item), pad-data(pin-item), side(pin-item), has-generic-props?(pin-item), has-power-props?(pin-item)])
-
-    val stm-supports = create-supports(input-json)
-    println("Here")
-
-main()
\ No newline at end of file
+public defn create-mcu-struct (file:String) -> STM :
+  val input-json = parse $ slurp(file)
```

Comment:
You need to add the necessary properties to MicroController in micro-controller.stanza,
adapt `public defn MicroController (json: JObject) -> MicroController :` and then adapt with the new properties (I don't think that the current version is used)
```
defmethod to-esir (mcu: MicroController) -> Instantiable :
  pcb-component my-microcontroller :
    ...
```

Then you delete `find-stm32-module` and
```
public defn micro-controller (params:Tuple<KeyValue>) :
  val mcu = MicroController(params)
  find-stm32-module(mpn(mcu))
```
becomes
```
public defn micro-controller (params:Tuple<KeyValue>) :
  val mcu = MicroController(params)
  to-esir(mcu)
```

This is the interface for database parts. It is the same with capacitors:
```
public defn Capacitor (raw-json: JObject) -> Capacitor :`
```

```
defmethod to-esir (capacitor: Capacitor) -> Instantiable :
  pcb-component my-capacitor :
```

```
public defn ceramic-cap (params:Tuple<KeyValue>) :
  val default-params = ["type" => "ceramic" 
                        "mounting" => "smd" 
                        "case" => get-valid-pkg-list()
                        "minimum_quantity" => 1
                        "min-rated-voltage" => MIN-CAP-VOLTAGE
                        "min-stock" => 5 * DESIGN-QUANTITY]
  to-esir(Capacitor(to-tuple $ to-hashtable<String, ?> $ cat(default-params params)))
```

Capacitor, inductor, Resistor, MicroController derive from Component
```
public deftype Component
public defmulti x (component: Component) -> Double
public defmulti y (component: Component) -> Double
public defmulti area (component: Component) -> Double
public defmulti to-esir (component: Component) -> Instantiable
public defmulti generate-esir (component: Component) -> Instantiable

defmethod area (component: Component) -> Double :
  x(component) * y(component)

defmethod generate-esir (cmp: Component) -> Instantiable :
  val esir-cmp = to-esir(cmp)
  ; ensure-generated(esir-cmp)
  esir-cmp
```
(I made `generate-esir`, in esir1 and didn't know how to adapt it to esir2 so it is useless)